databases/week3/WEEK3 homework/mealSharing_db

```diff
@@ -0,0 +1,256 @@
+
+
+-- Get a review with any id, fx 1
+
+SELECT *
+FROM Review
+WHERE id = 15;
+
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE Review
+SET description ='Food was good but customer service could improve , and price was a bit too much', stars=3
+WHERE title LIKE' Indian Brunch';
+
+
+
+-- Delete a review with any id, fx 1
+DELETE
+FROM Review
+WHERE title LIKE 'French Dinner';
+
+--Now add a couple of different meals, reservations and reviews 
+--with different attributes. With those meals create the
+--following queries
+
+
+
+
+
+
+INSERT INTO Meal(title,description,location,Hosting_time,max_reservations,price,created_date)
+Values('Ethiopian Injera and Tibs ','Injera, a traditional, fermented, spongy Ethiopian bread made with Teff flour','Addis Ababa, Ethiopia ','2022-09-10 17:50:20',10,50,'2022-07-20'),
+	(' Vegan Pesto Pasta ','Cavatelli Pasta-Fresh Pesto','Noto, Italy','2022-03-10 17:30:20',15,40,'2022-02-20'),
+    (' Samosa with Green Chutney','Samosa-Green Chutney (coriander dipping sauce)','Jaipur, India ','2022-04-10 17:30:20',20,30,'2022-03-20');
+
+
+
+        INSERT INTO Reservation(number_of_guests,meal_id,created_date,contact_phonenumber,contact_name,contact_email) 
+VALUES(2,7,'2022-07-28','44 67850238','Angi','Angi@databse.dk'),
+      (6,7,'2022-07-28','44 67850238','Saron','Sarii@databse.dk'),
+      (4,8,'2022-02-28','44 67850238','Maria','Mar@databse.dk'),
+      (7,9,'2022-04-28','44 67850238','Mongo','Monk@databse.dk'),
+      (8,9,'2022-04-28','44 67850238','Mik','Mikke@databse.dk');
+
+
+      INSERT INTO Review(title,description,meal_id,stars,created_date)
+VALUES('Ethiopian Injera'," We’ll definitely do this again",7,4,'2021-02-26'),
+       ('Ethiopian food',"We had a wonderful time, learned about Ethiopian food and culture and had an amazing meal",7,5,'2021-02-26'), 
+       ('Injera food',"Thank you so much for this nice experience",7,5,'2021-02-26'), 
+       ('Vegan Pesto Pasta ',"It was a fun and unique experience! Thank you",8,4,'2021-02-26'), 
+       (' Pesto Pasta '," I believe my white flour would have been better.",8,3,'2021-02-26'), 
+       ('Vegan Pesto',"We had a great time. They were fantastic hosts ",8,4,'2021-02-26'), 
+       ('Samosa with Green Chutney',"I got it as a gift for my daughter and her boyfriend because they love Smosa",9,5,'2021-02-26'), 
+       ('Samosa ',"great meal",9,5,'2021-02-26'), 
+       ('Samosa with Chutney'," an amazing experience and we’ll be back for sure.",9,5,'2021-02-26');
+
+
+    --Functionality
+-- Get meals that has a price smaller than a specific price fx 90
+
+SELECT*
+ FROM Meal
+ WHERE price <50;
+-- Get meals that still has available reservations
+Select SUM(number_of_guests) AS reserved,meal_id,Meal.title,max_reservations
+FROM Reservation 
+JOIN Meal On Meal.id= Reservation.meal_id
+-- Where Reservation.reserved < Meal.max_reservations
```

Comment:
You're almost there! :D
As you probably noticed, the WHERE doesn't work with GROUP BY, you need to use HAVING instead.
Then don't forget to sum the number of guests of all reservations and compare with the max reservations :)

Another, slightly more complicated option is to use the COALESCE() function: https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_coalesce

Reply:
I don't get why do I need to  "sum the number of guests of all reservations and compare with the max reservations". I got question as it wants us to return all the food which are not totally booked.

Reply:
Yes, and each meal can have many reservations, and each reservation has a certain number of guests associated with it. so you need to know what is the total number of guests for a given meal, to see if you can still make more reservations.

Let's say we have meal 1 with 3 reservations: reservation 1 has 2 guests, reservation 2 has 10 guests, and reservation 3 has 4 guests. There are a total of 16 guests for meal 1. If meal 1 has max_reservations of 20, you can still accept 4 guests.
Does this make more sense now?

Reply:
yes , I understand it now .